dotCMS/src/main/java/com/dotcms/webdav/FsResource.java

```diff
@@ -0,0 +1,162 @@
+import com.dotcms.repackage.com.bradmcevoy.http.MoveableResource;
+import com.dotcms.repackage.com.bradmcevoy.http.Request;
+import com.dotcms.repackage.com.bradmcevoy.http.Request.Method;
+import com.dotcms.repackage.com.bradmcevoy.http.Resource;
+import com.dotcms.repackage.com.bradmcevoy.http.exceptions.NotAuthorizedException;
+import com.dotcms.repackage.com.bradmcevoy.http.http11.auth.DigestResponse;
+import com.dotmarketing.util.Logger;
+
+
+import java.io.File;
+import java.util.Date;
+
+
+/**
+ *
+ */
+public abstract class FsResource implements Resource, MoveableResource, CopyableResource, LockableResource, DigestResource {
+
+
+    File file;
+    final FileSystemResourceFactory factory;
+    final String host;
+    String ssoPrefix;
+
+    protected abstract void doCopy(File dest);
+
+    public FsResource(String host, FileSystemResourceFactory factory, File file) {
+        this.host = host;
+        this.file = file;
+        this.factory = factory;
+    }
+
+    public File getFile() {
+        return file;
+    }
+
+    public String getUniqueId() {
+        String s = file.lastModified() + "_" + file.length() + "_" + file.getAbsolutePath();
+        return s.hashCode() + "";
+    }
+
+    public String getName() {
+        return file.getName();
+    }
+
+    public Object authenticate(String user, String password) {
+        return factory.getSecurityManager().authenticate(user, password);
+    }
+
+    public Object authenticate(DigestResponse digestRequest) {
+        return factory.getSecurityManager().authenticate(digestRequest);
+    }
+
+    public boolean isDigestAllowed() {
+        return factory.isDigestAllowed();
+    }
+
+    public boolean authorise(Request request, Method method, Auth auth) {
+        boolean b = factory.getSecurityManager().authorise(request, method, auth, this);
+        Logger.info(FileSystemResourceFactory.class,"authorise: result=" + b);
+
+        return b;
+    }
+
+    public String getRealm() {
+        return factory.getRealm(this.host);
+    }
+
+    public Date getModifiedDate() {
+        return new Date(file.lastModified());
+    }
+
+    public Date getCreateDate() {
+        return null;
+    }
+
+    public int compareTo(Resource o) {
+        return this.getName().compareTo(o.getName());
+    }
+
+    public void moveTo(CollectionResource newParent, String newName) {
+        if (newParent instanceof FsDirectoryResource) {
+            FsDirectoryResource newFsParent = (FsDirectoryResource) newParent;
+            File dest = new File(newFsParent.getFile(), newName);
+            boolean ok = this.file.renameTo(dest);
+            if (!ok) {
+                throw new RuntimeException("Failed to move to: " + dest.getAbsolutePath());
+            }
+            this.file = dest;
+        } else {
+            throw new RuntimeException("Destination is an unknown type. Must be a FsDirectoryResource, is a: " + newParent.getClass());

```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Define and throw a dedicated exception instead of using a generic one. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00112)